src/api/blockchains/tokens/pipes/address-validation.pipe.ts

```diff
@@ -0,0 +1,29 @@
+import { BadRequestException, Injectable, PipeTransform } from '@nestjs/common';
+import { ConfigService } from '@nestjs/config';
+
+import { Web3Provider } from '../web3.provider';
+
+@Injectable()
+export class AddressValidationPipe implements PipeTransform {
+  constructor(private readonly configService: ConfigService) {}
+  async transform(value: { chainId: string; address: string }) {
+    const { chainId, address } = value;
+    let isContract: boolean;
+
+    try {
+      isContract = await new Web3Provider(
+        this.configService,
+        Number(chainId),
+      ).isContract(address);
+    } catch (error) {
+      throw new BadRequestException('Address is not a valid Ethereum address');
```

Comment:
log error

Reply:
> Error may be thrown due to infura configuration etc., exception description is not accurate

I've moved the `await new Web3Provider(...)` statement outside of the `try...catch` blocks, if it throws an error it will not be caught.